libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -266,6 +267,68 @@ std::string YulUtilFunctions::requireOrAssertFunction(bool _assert, Type const*
 	});
 }
 
+std::string YulUtilFunctions::requireWithError(FunctionCall const& errorConstructorCall)
+{
+	ErrorDefinition const* errorDefinition = dynamic_cast<ErrorDefinition const*>(ASTNode::referencedDeclaration(errorConstructorCall.expression()));
+	solAssert(errorDefinition);
+
+	std::string const errorSignature = errorDefinition->functionType(true)->externalSignature();
+	std::string const signatureHash = formatNumber(util::selectorFromSignatureU256(errorSignature));
```

Comment:
The selector can easily collide... we do prevent that in analysis (see e.g. test/libsolidity/syntaxTests/errors/hash_collision.sol), but it may still be better to use the errorDefinition's AST ID instead (see e.g. ``IRNames::function`` in ``libsolidity/codegen/ir/Common.cpp`` and such)

Reply:
Yeah, at the time of that comment, the function name was "disambiguated" using the selector, that's what the comment aimed at.